beluga-mqtt/src/lib.rs

```diff
@@ -165,47 +180,24 @@ impl MqttClient {
     /// # Returns
     /// A [Result] containing a [Subscriber] if the subscriptions are
     /// successful, otherwise an [Error].
-    pub async fn subscribe_many(
+    pub async fn subscribe_many<'a>(
         &self,
-        topics: impl IntoIterator<Item = &str>,
+        topics: impl IntoIterator<Item = &'a str> + 'a,
```

Comment:
We might as well make topics generic over `AsRef<str>`, right?
As in `where It: IntoIterator, It::Item: AsRef<str>`.
That way you don't need the `as_str()` when using `format!()`.